src/tests/mod.rs

```diff
@@ -487,21 +487,44 @@ mod stash {
     }
 
     #[test]
-    pub(crate) fn stash_working_tree_then_index_and_then_pop() {
+    pub(crate) fn stash_working_tree_when_everything_is_staged() {
         let (mut ctx, mut state) = setup();
         state
             .update(
                 &mut ctx.term,
                 &[
+                    key('j'),
+                    key('s'),
                     key('z'),
                     key('w'),
+                    key('t'),
+                    key('e'),
+                    key('s'),
+                    key('t'),
                     key_code(KeyCode::Enter),
+                ],
+            )
+            .unwrap();
+        insta::assert_snapshot!(ctx.redact_buffer());
+    }
+
+    #[test]
+    pub(crate) fn stash_working_tree_when_nothing_is_staged() {
+        let (mut ctx, mut state) = setup();
+        state
+            .update(
+                &mut ctx.term,
+                &[
+                    key('j'),
+                    key('j'),
+                    key('j'),
+                    key('u'),
                     key('z'),
-                    key('i'),
-                    key_code(KeyCode::Enter),
-                    key('z'),
-                    key('p'),
-                    key('1'),
+                    key('w'),
+                    key('t'),
+                    key('e'),
+                    key('s'),
+                    key('t'),
```

Comment:
I'd love to solve the syntax for sending these events somehow.
Would need one for keybind config anyway.
Something like, "jjjuzwtest<enter>".
And perhaps a test helper fn to select lines:
`select("Unstaged changes.*");`

Doesn't belong in this PR though.

Reply:
Wow, this sounds awesome! Implementing "jjjuzwtest<enter>" doesn't sound very hard tho. Selecting on the other hand looks pretty hard in my head, since we have everything inside the terminal. Does it have something like `read_to_string`? I need to research it a little bit

Reply:
It's all in an in-memory buffer for tests. Seems a bit hard to work with.